File: spring-boot-dubbo/dubbo-client/src/main/java/com/xphsc/dubbo/client/controller/UserController.java
package com.xphsc.dubbo.client.controller;

import com.alibaba.dubbo.config.annotation.Reference;
import com.github.xphsc.http.Response;
import com.xphsc.dubbo.api.model.response.UserDTO;
import com.xphsc.dubbo.api.service.UserService;
import org.springframework.web.bind.annotation.GetMapping;
import org.springframework.web.bind.annotation.RestController;

import java.util.List;

/**
 * Created by ${huipei.x} on 2017-2-25.
 * qq群593802274
 */
@RestController
public class UserController {

    @Reference
    private UserService userService;

    @GetMapping("/listUser")
    public Response listUser(){
        List<UserDTO> result= userService.listUser();
        return Response.successResult(result);
    }

}

File: spring-boot-kotlin/src/main/java/com/xphsc/kotlin/controller/UserController.kt
package com.xphsc.kotlin.controller

import com.github.xphsc.http.Response
import com.xphsc.kotlin.model.response.UserDTO
import com.xphsc.kotlin.service.UserService
import org.springframework.web.bind.annotation.GetMapping
import org.springframework.web.bind.annotation.RestController

/**
 * Created by ${huipei.x} on 2017-2-25
 */
@RestController
 class UserController(private val userService: UserService) {

    @GetMapping("/listUser")
    fun listUser():Response{
      val result : List<UserDTO>  =userService.listUser();
        return Response.successResult(result);
    }
}